web-console/src/views/query-view/query-view.tsx

```diff
@@ -384,11 +411,60 @@ export class QueryView extends React.PureComponent<QueryViewProps, QueryViewStat
             )}
           </div>
         </div>
-        <QueryOutput loading={loading} result={result} error={error} />
+        <QueryOutput
+          aggregateColumns={ast ? ast.getAggregateColumns() : undefined}
+          disabled={!ast}
+          sorted={ast ? ast.getSorted() : undefined}
+          handleSqlAction={this.handleSqlAction}
+          loading={loading}
+          result={result}
+          error={error}
+        />
       </SplitterLayout>
     );
   }
 
+  private handleSqlAction = (row: string, header: string, action: string): void => {
```

Comment:
maybe worth breaking this one down so there is no switch statement, the way it is coded now seems like an anti-pattern